core/src/main/java/com/testaem/aem/core/models/impl/SinglePageNewsImpl.java

```diff
@@ -50,4 +110,24 @@ public String getImagePath() {
     public String getDate() {
         return this.date == null ? simpleDateFormat.format(new Date()) : simpleDateFormat.format(date);
     }
+
+    public String getPagePath() {
+
+        List<String> selectors = Arrays.asList(request.getRequestPathInfo().getSelectors());
+
+        boolean fullPage = selectors.contains("fullPage");
+
+        PageManager pageManager = request.getResourceResolver().adaptTo(PageManager.class);
+        Page containingPage = pageManager.getContainingPage(request.getResource());
+        ;
+        Page landingPage = containingPage.getParent(2);
+
+        String path = fullPage ? containingPage.getPath() : landingPage.getPath();
```

Comment:
(cosmetic) You see why good names are important. The statement looks like there are three pages: fullPage, containingPage, and landingPage. And they are all somewhat related) But actually there are two.

Why not renaming the boolean to `isLandingPageRequested` or sorts?)